test/generators/typescript/preset/MarshallingPreset.spec.ts

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable */
+
+import { TypeScriptGenerator, TS_COMMON_PRESET } from '../../../../src/generators'; 
+import Ajv from 'ajv';
+const doc = {
+  $id: 'Test',
+  type: 'object',
+  additionalProperties: true,
+  properties: {
+    stringProp: { type: 'string' },
+    numberProp: { type: 'number' },
+    objectProp: { type: 'object', $id: 'NestedTest', properties: {stringProp: { type: 'string' }}}
+  },
+  patternProperties: {
+    '^S(.?)test': {
+      type: 'string'
+    }
+  },
+};
+describe('TS_COMMON_PRESET', () => {
+  test('should render un/marshal code', async () => {
+    const generator = new TypeScriptGenerator({ 
+      presets: [
+        {
+          preset: TS_COMMON_PRESET,
+          options: {
+            marshal: true,
+            unmarshal: true
+          }
+        }
+      ]
+    });
+    const inputModel = await generator.process(doc);
+    const testModel = inputModel.models['Test'];
+    const nestedTestModel = inputModel.models['NestedTest'];
+
+    const testClass = await generator.renderClass(testModel, inputModel);
+    const nestedTestClass = await generator.renderClass(nestedTestModel, inputModel);
+
+    expect(testClass.result).toMatchSnapshot();
+    expect(nestedTestClass.result).toMatchSnapshot();
+  });
+
+  test('should provide a two way conversion', async () => {
+    class NestedTest {
+      private _stringProp?: string;
+      private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null | number>;
+    
+      constructor(input: {
+        stringProp?: string,
+      }) {
+        this._stringProp = input.stringProp;
+      }
+    
+      get stringProp(): string | undefined { return this._stringProp; }
+      set stringProp(stringProp: string | undefined) { this._stringProp = stringProp; }
+    
+      get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined { return this._additionalProperties; }
+      set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null | number> | undefined) { this._additionalProperties = additionalProperties; }
+    
+      public marshal() : string {
+        let json = '{'
+        if(this.stringProp !== undefined) {
+          json += `"stringProp": ${JSON.stringify(this.stringProp)},`; 
+        }
+    
+      
+    
+        if(this.additionalProperties !== undefined) { 
+          for (const [key, value] of this.additionalProperties.entries()) {
+            //Only render additionalProperties which are not already a property
+            if(Object.keys(this).includes(String(key))) continue;
+            json += `"${key}": ${JSON.stringify(value)},`;    
+          }
+        }
+        return `${json.charAt(json.length-1) === ',' ? json.slice(0, json.length-1) : json}}`;
+      }
```

Comment:
Did you think about solution when you create based on given instance the normal JS object with keys and at the end make `JSON.stringify`, so you operate on object (which should be easier to handle) not on string?

Reply:
That's actually a good point, I wonder if there are any performance improvements to be had there too 🤔? Or if string manipulation is faster.

Reply:
I wouldn't mind changing it tbh, as it would improve readability I think. But not if we half the performance 😆

Reply:
Maybe performance will be worse, but then you don't handle manually cases such as https://github.com/asyncapi/modelina/pull/315#discussion_r685813405 so object would improve readability :)

Reply:
Hmm... I think we keep it as is for now, in the future, we can always address the readability, as un/marshalling will be one of the main executed functions for many libraries, I would opt for performance > readability. What do you think @magicmatatjahu ?

Reply:
ok :) Could you create issue for that and point to our discussion? So then, remember about this https://github.com/asyncapi/modelina/pull/315#discussion_r685813405 case.

Reply:
I can yes, and yea I will remember that comment, I will go through all before pushing changes, so it will be addressed 😄

Reply:
https://github.com/asyncapi/modelina/issues/317